[BUGFIX lts-3-12] Upgrade glimmer-vm to v0.38.5-alpha.1.6.1

chancancode opened the pull request:
Diff: https://github.com/glimmerjs/glimmer-vm/compare/v0.38.5-alpha.1...v0.38.5-alpha.1.6.1

This is essentially backporting all the bug fixes in versions up to v0.38.5-alpha.6 but without the potentially breaking patches that affects how tags work.

In addition to the changes described in the CHANGELOG, this also included a fix for #18184 but there were no specific CHANGELOG entry for it that I could backport.

It also included an upgraded version of simple-html-tokenizer that parses named blocks syntax correctly, making it more feasible to polyfill the feature.

As far as I can tell, there are no tests that I could find on lts-3-16 that I should backport.

chancancode committed:
[BUGFIX lts-3-12] Upgrade glimmer-vm to v0.38.5-alpha.1.6.1

Diff: https://github.com/glimmerjs/glimmer-vm/compare/v0.38.5-alpha.1...v0.38.5-alpha.1.6.1

This is essentially backporting all the bug fixes in versions up to
v0.38.5-alpha.6 but without the potentially breaking patches that
affects how tags work.

In addition to the changes described in the CHANGELOG, this also
included a fix for #18184 but there were no specific CHANGELOG
entry for it that I could backport.

It also included an upgraded version of simple-html-tokenizer that
parses named blocks syntax correctly, making it more feasible to
polyfill the feature.

As far as I can tell, there are no tests that I could find on
lts-3-16 that I should backport.

chancancode commented on yarn.lock:
```diff
@@ -668,122 +668,160 @@
+    "@glimmer/util" "^0.38.5-alpha.1.6"
+
+"@glimmer/reference@^0.38.5-alpha.1.6", "@glimmer/reference@^0.38.5-alpha.6":
+  version "0.38.5-alpha.6"
```
??????

rwjblue replied:
WAT, does yarn not know how to deal with this?!

rwjblue commented on package.json:
```diff
@@ -93,14 +93,14 @@
     "@babel/plugin-transform-spread": "^7.2.2",
     "@babel/plugin-transform-template-literals": "^7.4.4",
-    "@glimmer/compiler": "0.38.5-alpha.1",
+    "@glimmer/compiler": "0.38.5-alpha.1.6",
```
LOL @ this version number :sob: :joy:

chancancode commented on yarn.lock:
```diff
@@ -668,122 +668,160 @@
-  resolved "https://registry.yarnpkg.com/@glimmer/syntax/-/syntax-0.38.5-alpha.1.tgz#5b2c13aff088a39c326e142b465c57e7e3cf8ba9"
-  integrity sha512-VB5mQFqL2A9y8Sx9spALarokyZRfRzgClsmX+K3J19N+oLuXu8al9+5T9utDTLDZy33PQzFMkDYf/x91n3w7ow==
+"@glimmer/runtime@^0.38.5-alpha.1.6":
+  version "0.38.5-alpha.6"
```
????????????????

chancancode replied:
Did we break yarn?

chancancode commented on package.json:
```diff
@@ -172,5 +172,20 @@
+    "@glimmer/syntax": "0.38.5-alpha.1.6",
+    "@glimmer/util": "0.38.5-alpha.1.6",
+    "@glimmer/vm": "0.38.5-alpha.1.6",
+    "@glimmer/wire-format": "0.38.5-alpha.1.6"
```
Without these, yarn seems to be pretty liberal when interpreting the "^0.38.5-alpha.1.6" in these package's dependencies list. Should I commit this or not?

rwjblue replied:
Ya, seems fine to do here, but we could also publish a `0.38.0-alpha.1.6.1` that depends on exact versions (instead of `^`) which probably would also solve the issue.

These resolutions seem easier to me though...

chancancode replied:
`0.38.0-alpha.1.6.1` 😭

rwjblue commented on package.json:
```diff
@@ -93,14 +93,14 @@
     "@babel/plugin-transform-spread": "^7.2.2",
     "@babel/plugin-transform-template-literals": "^7.4.4",
-    "@glimmer/compiler": "0.38.5-alpha.1",
+    "@glimmer/compiler": "0.38.5-alpha.1.6.1",
```
LOLOLOL

chancancode committed:
Remove existing `node_modules` before install

We are having issues on the lts-3-12 branch, when reusing the cache from
master, `yarn install` correctly patches `node_modules/@babel/core` to
7.4.5, but fails to remove `node_modules/@babel/core/node_modules`,
which contains `@babel/types` 7.9.6, which is not what we want and is
inconsistent with the lockfile.

Since we are not goint to get good cache reuse from master's builds
anyway, and that lts-3-12 gets build so rarely it is unlikely to have a
cache available, the safest thing to do for now seems to be starting
from a clean `node_modules` folder. We will still benefit from having
the yarn cache folder available to us and therefore save on network
fetches.
